functional-tests/factory.py

```diff
@@ -0,0 +1,313 @@
+import os
+from math import ceil
+from typing import Optional, TypedDict
+
+import flexitest
+import web3
+import web3.middleware
+from bitcoinlib.services.bitcoind import BitcoindClient
+
+import seqrpc
+import net_settings
+from constants import *
+from utils import *
+
+
+class BitcoinRpcConfig(TypedDict):
+    bitcoind_sock: str
```

Comment:
This can be to be consistent with the `RethConfig`?

```suggestion
    bitcoind_socket: str
```

Reply:
Yeah the bridge client needs to be made more consistent with the rollup client too.

Reply:
I don't want to change too much and break anything because I didn't write any of this code.

Reply:
Actually I can't figure out anywhere that `BitcoinRpcConfig` is constructed, I'm not sure how to use this type.